src/app/app.component.ts

```diff
@@ -198,35 +207,84 @@ export class AppComponent implements AfterViewInit {
+        isLeaf: false,
+        path: this.selectedFolder.path,
+        name: this.selectedFolder.displayName,
+        icon: ''
+      }
+    } else {
+      focusedNode = defaultNode;
+    }
+    await this.lfRepositoryBrowser?.nativeElement.initAsync(this.lfRepoTreeNodeService, focusedNode);
+  }
+
+  isNodeSelectable = (node: LfRepoTreeNode) => {
+    if (node.entryType == EntryType.Folder) {
+      return true
+    }
+    else {
+      return false
+    }
   }
 
   get isLoggedIn(): boolean {
     return this.loginComponent?.nativeElement?.state === LoginState.LoggedIn;
   }
 
   // Tree event handler methods
-  async onOkClick(okClickEvent: Event) {
-    const selectedNode = (okClickEvent as CustomEvent<TreeNode>).detail;
-    const breadcrumbs = this.lfFolderBrowserElement?.nativeElement?.breadcrumbs;
-    const entryId = Number.parseInt(selectedNode.id, 10);
-    const path = selectedNode.path;
-    this.selectedFolder = {
-      entryId,
-      path,
-      displayName: this.getFolderNameText(entryId, path),
-      displayPath: this.getFolderPathTooltip(path)
-    };
-    if (breadcrumbs) {
-      this.selectedFolder.breadcrumbs = breadcrumbs;
-    }
-    const nodeId = selectedNode.id;
-    const repoId = (await this.repoClient.getCurrentRepoId());
-    const waUrl = this.loginComponent.nativeElement.account_endpoints.webClientUrl;
-    this.selectedNodeUrl = getEntryWebAccessUrl(nodeId, repoId, waUrl, selectedNode.isContainer);
-    this.expandFolderBrowser = false;
+  async onSelectFolder() {
+    const selectedNode = this.lfRepositoryBrowser.nativeElement.currentFolder as LfRepoTreeNode;
+
+
+      const breadcrumbs = this.lfRepositoryBrowser?.nativeElement?.breadcrumbs;
+      let entryId = Number.parseInt(selectedNode.id, 10);
+      const path = selectedNode.path;
+      if (selectedNode.entryType == EntryType.Shortcut) {
+        entryId = selectedNode.targetId;
+      }
+      this.selectedFolder = {
+        entryId,
+        path,
+        displayName: this.getFolderNameText(entryId, path),
+        displayPath: this.getFolderPathTooltip(path)
+      };
+      const nodeId = selectedNode.id;
+      const repoId = (await this.repoClient.getCurrentRepoId());
+      const waUrl = this.loginComponent.nativeElement.account_endpoints.webClientUrl;
+      this.selectedNodeUrl = getEntryWebAccessUrl(nodeId, repoId, waUrl, selectedNode.isContainer);
+      this.expandFolderBrowser = false;
+  }
+
+  get shouldShowSelect(): boolean {
+    return !this.shouldShowOpen && !!this.lfRepositoryBrowser?.nativeElement?.currentFolder;
+  }
+
+  get shouldShowOpen(): boolean {
+    return !!this.entrySelected;
+  }
+
+  onEntrySelected(event) {
+    const customEvent = event as CustomEvent<LfTreeNode[]>;
+    const treeNodesSelected: LfTreeNode[] = customEvent.detail;
+    this.entrySelected = treeNodesSelected?.length>0 ? treeNodesSelected[0] : undefined;
+  }
+
+  async onOpenNode() {
+    const nodeToOpen = this.entrySelected;
```

Comment:
hmm this wouldn't work on the element anyways... but maybe it does through the npm package?
Either way this isn't the function we want to use